flow/layers/opacity_layer.h

```diff
@@ -38,8 +38,42 @@ class OpacityLayer : public ContainerLayer {
 #endif  // defined(OS_FUCHSIA)
 
  private:
+  /**
+   * @brief Returns the ContainerLayer used to hold all of the children
+   * of the OpacityLayer.
+   *
+   * Typically these layers should only have a single child according to
+   * the contract of the associated Flutter widget, but the engine API
+   * cannot enforce that contract and must be prepared to support the
+   * possibility of adding more than one child. In either case, the
+   * child container will hold all children. Note that since the child
+   * container is created fresh with each new OpacityLayer, it may not
+   * be the best choice to cache to speed up rendering during an animation.
```

Comment:
"Typically these layers should only have a single child according to the contract of the associated Flutter widget" is probably inaccurate: although `Opacity` widget has a single child, that child may not `needsCompositing` and therefore won't create an engine layer, and its children may end up as the direct children of `OpacityLayer`.

This triggers me to think should we always automatically add a `RepaintBoundary` as the child of an `Opacity` widget so an `Opacity` widget would always create an `OpacityLayer` that has a single `ContainerLayer` child. Without it, the Opacity widget may end up having 2 children that are unchanged across frames, and even the fix here won't be able to re-use the layer raster cache?

Reply:
I thought that RepaintBoundary was the only widget that could add a retained layer, so without it the children change anyway whether or not they have `needsCompositing`...?

Reply:
Ah yes, without `RepaintBoundary` they'll not be retained. Anyway, the `Opacity` widget has no contract of a single child layer (it may have multiple layers that are not retained across frames). LGTM once that statement on the contract is revised!

Reply:
BTW, maybe add a comment like "Even if `GetCacheableChild()` is changing in each frame and we cannot reuse the layer raster cache, the single child exposed by `GetChildContainer()` allows us to paint the child in `Preroll` onto an offscreen buffer; it reduces one render target switch (compared to painting the child in `Paint`) and thus still improves our performance."

Reply:
I'm updating the comment.

Reply:
I verified with my standalone version of the benchmark that if I remove the RepaintBoundary wrapping the children, we still end up with one child of the OpacityLayer, but the performance tanks because it is a different child each time. So, RPB is pretty much needed for performance anyway and it ensures the single child condition.

Reply:
New version of the comment is now pushed.

Reply:
I didn't see your comment about the fallback issue before the last push, but I added a paragraph about that in the latest push.